### PR TITLE
Interaction message flags

### PR DESCRIPTION
**Pull request by HopeBaron**
Followups required some changes on how to deal with followups based on the following scenarios:

(E) ACK + Followup = (E) Follow up on the same message
(P) ACK + Followup = (P) Followup on same message
(E) Respond + Followup = (P) Followup
(P) Respond + Followup = (P) Followup

(P): Public
(E): Ephemeral

Also this PR aims to provide a better interface over ephmerals/public responses.

### CODE REVIEW

**Comment by BartArys** [comment]
> 
> 
> Followups required some changes on how to deal with followups based on the following scenarios:
> 
> (E) ACK + Followup = (E) Follow up on the same message
> (P) ACK + Followup = (P) Followup on same message
> (E) Respond + Followup = (P) Followup
> (P) Respond + Followup = (P) Followup
> 
> (P): Public
> (E): Ephemeral
> 
> Also this PR aims to provide a better interface over ephmerals/public responses.

what in tarnation?

**Comment by gdude2002** [comment]
> what in tarnation?

image

File: core/src/main/kotlin/entity/interaction/Interaction.kt [hunk 1]
```diff
@@ -49,10 +52,6 @@ sealed class Interaction : InteractionBehavior {
      */
     val type: InteractionType get() = data.type
 
-    /**
-     * The [MessageChannelBehavior] of the channel the command was executed in.
-     */
-    open val channel: MessageChannelBehavior get() = MessageChannelBehavior(data.channelId, kord)
```
Comment (BartArys) [inline core/src/main/kotlin/entity/interaction/Interaction.kt:55]: An interaction is always in a message channel, no?

File: core/src/main/kotlin/event/interaction/InteractionCreate.kt [hunk 2]
```diff
@@ -2,9 +2,8 @@ package dev.kord.core.event.interaction
 
 import dev.kord.common.annotation.KordPreview
 import dev.kord.core.Kord
-import dev.kord.core.entity.interaction.Interaction
 import dev.kord.core.event.Event
-
+import dev.kord.core.entity.interaction.Interaction
 @KordPreview
```
Comment (BartArys) [inline core/src/main/kotlin/event/interaction/InteractionCreate.kt:7]:
```suggestion
import dev.kord.core.entity.interaction.Interaction

@KordPreview
```

File: core/src/main/kotlin/behavior/interaction/EphemeralInteractionResponseBehavior.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+@KordPreview
+interface EphemeralInteractionResponseBehavior : InteractionResponseBehavior
+
+@KordPreview
+@OptIn(ExperimentalContracts::class)
+suspend inline fun EphemeralInteractionResponseBehavior.edit(builder: EphemeralInteractionResponseModifyBuilder.() -> Unit) {
+    contract { callsInPlace(builder, InvocationKind.EXACTLY_ONCE) }
+    val builder = EphemeralInteractionResponseModifyBuilder().apply(builder)
+    kord.rest.interaction.modifyInteractionResponse(applicationId, token, builder.toRequest())
+}
```
[the remark this answers is not in the document]
Comment (HopeBaron) [reply core/src/main/kotlin/behavior/interaction/EphemeralInteractionResponseBehavior.kt:39]: Confirmed, it does return a correct Message object based on the data debugged.

**Review by BartArys — APPROVED** [review]

**Pull request merged**